app/models/api_key.rb

```diff
@@ -48,13 +48,26 @@ def enabled_scopes
     end
   end
 
+  def user
+    owner if user?
+  end
+
+  def user?
+    owner_type == "User"
+  end
+
+  delegate :mfa_required_not_yet_enabled?, :mfa_required_weak_level_enabled?,
+    :mfa_recommended_not_yet_enabled?, :mfa_recommended_weak_level_enabled?,
+    to: :user, allow_nil: true
+
```

Comment:
Just an idea that might not improve the code: If you added a second delegate with prefix: true for user_mfa_enabled? and others, it might make some of those easier.

Reply:
I'm not sure exactly what benefit that would have?